sdk/python/feast/client.py

```diff
@@ -635,14 +642,37 @@ def get_online_features(
             response = self._serving_service_stub.GetOnlineFeatures(
                 GetOnlineFeaturesRequest(
                     features=_build_feature_references(
-                        feature_refs=feature_refs,
-                        default_project=(
-                            default_project if not self.project else self.project
-                        ),
+                        feature_ref_strs=feature_refs,
+                        project=project if project is not None else self.project,
                     ),
                     entity_rows=entity_rows,
                 )
             )
+            # collect entity row refs
+            entity_refs = set()
+            for entity_row in entity_rows:
+                entity_refs.update(entity_row.fields.keys())
+
+            def strip_project(field_value):
```

Comment:
What is the advantage of nesting this method as opposed to in-lining the code in a loop or moving the method out? `get_online_features` is already quite big, so perhaps we should handle the response cleanup elsewhere.

Reply:
Removed nesting in favor of a loop.